crates/wasmprinter/src/lib.rs

```diff
@@ -2577,6 +2584,73 @@ impl Printer {
         self.end_group();
         Ok(())
     }
+
+    fn print_dylink0_section(&mut self, mut section: Dylink0SectionReader<'_>) -> Result<()> {
+        self.start_group("@dylink.0");
+        loop {
+            let start = section.original_position();
+            let next = match section.next() {
+                Some(Ok(next)) => next,
+                Some(Err(e)) => return Err(e.into()),
+                None => break,
+            };
+            match next {
+                Dylink0Subsection::MemInfo(info) => {
+                    self.newline(start);
+                    self.start_group("mem-info");
+                    if info.memory_size > 0 || info.memory_alignment > 0 {
+                        write!(
+                            self.result,
+                            " (memory {} {})",
+                            info.memory_size, info.memory_alignment
+                        )?;
+                    }
+                    if info.table_size > 0 || info.table_alignment > 0 {
+                        write!(
+                            self.result,
+                            " (table {} {})",
+                            info.table_size, info.table_alignment
+                        )?;
+                    }
+                    self.end_group();
+                }
+                Dylink0Subsection::Needed(needed) => {
+                    self.newline(start);
+                    self.start_group("needed");
+                    for s in needed {
+                        self.result.push_str(" ");
+                        self.print_str(s)?;
+                    }
+                    self.end_group();
+                }
+                Dylink0Subsection::ExportInfo(info) => {
+                    for info in info {
+                        self.newline(start);
+                        self.start_group("export-info ");
+                        self.print_str(info.name)?;
+                        write!(self.result, " {}", info.flags)?;
```

Comment:
Similar to the above, it would be cool if we could print and parse the flags symbolically (e.g. `(binding-weak binding-local)` meaning `WASM_SYM_BINDING_WEAK | WASM_SYM_BINDING_LOCAL`; not sure what the "right" way to represent flags as s-expressions would be).  Not a blocker, of course.

Reply:
nah I was thinking the same thing! I was thinking `(export-info "x" binding-weak binding-local 0x80)` where you'd or-together everything after the string `"x"` to represent this, and it could handle unknown flags in theory too with integer literals

Reply:
I went ahead an implemented this, but I'll leave the bitflags bits for later